### PR TITLE
PedCBio formatted sample ID column

### PR DESCRIPTION
**Pull request by runjin326**
#### What scientific question is your analysis addressing?

Currently, for some of the samples, when multiple DNA or RNA specimens are associated with the same sample, there 
is no column that would distinguish between different aliquots while still tieing DNA and RNA together.

This PR adds a module to address that question.

#### What was your approach?
1. Some samples already have formatted ID in DWH - and for those, I just use that file to populate the column 
2. For samples that do not have formatted ID in DWH - I generate them based on cohorts
    - For PBTA samples, `sample_id` column is used as the basename
        - Multiple DNA samples associated with the same sample would use `aliquot_id` as the tie breaker
        - Multiple RNA samples associated with the same sample would use `RNA_library` as the tie breaker 
    - For TCGA, TARGET and GTEx samples, `Kids_First_Participant_ID` column is used as the basename
        - For TARGET, `Kids_First_Participant_ID` + last 7 digits from the `Kids_First_Specimen_ID` is used as formatted sample ID
        - For TCGA, `Kids_First_Participant_ID` + `sample_id` + `aliquot_id` is used as formatted sample ID
        - For GTEx, `Kids_First_Participant_ID` + `aliquot_id` is used as formatted sample ID

#### What GitHub issue does your pull request address?
https://github.com/PediatricOpenTargets/ticket-tracker/issues/263

#### Which areas should receive a particularly close look?
No.

#### Is there anything that you want to discuss further?
No.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes.

#### What types of results are included (e.g., table, figure)?
`results/histologies-formatted-id-added.tsv`

#### What is your summary of the results?
For samples with multiple DNA or RNA specimens associated with them, individual specimens are now distinguishable through formatted sample ID.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/pedcbio-sample-name/pedcbio_sample_name_col.Rmd [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+```{r load library}
+library(tidyverse)
+library(readr)
+```
+
+### Define directories
+```{r define directories}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
```
Comment (migbro) [inline analyses/pedcbio-sample-name/pedcbio_sample_name_col.Rmd]: Can this and perhaps the shell script be made to take optional args? Having defaults is fine, but if I run this in the current branch, there is no existing histologies file in the `data` dir

Comment (runjin326) [reply analyses/pedcbio-sample-name/pedcbio_sample_name_col.Rmd]: Hey @migbro - this script assumes that this is ran from OpenPedCan GitHub repo and the repo should have a `data` directory in the root directory. We do need to run the `download_data.sh` to get the data from S3 bucket and that is probably why you do not have the file in the data directory. 
That being said, I now updated the bash script and now you can go into the script, modify the input dir ($hist_dir) and run the bash script. We normally do not write the bash script that would take additional parameters through command line so you would have to modify within the script unfortunately.

**Comment by runjin326** [comment]
Thanks so much for the review - I have now also fixed the TCGA file names - please double check to see whether they are all unique now.

**Comment by migbro** [comment]
@runjin326 , ok so Adam R. reached out to me, and he seems keen on seeing the project loaded. What will unblock this is that he has given input on preference to the issue of matched TARGET T/N RNA samples. He's like both loaded - so please add on more ability to differentiate between the 2 - a suggestion would be adding a `-T` and `-N`, but if there is already something easy that can help distinguish, feel free to use that. Feel free to reach out if you need clarification!

**Comment by runjin326** [comment]
@migbro - thanks for the input - I would work on that as soon as I can. Have a couple things to work on before that but would be able to get to this on Wednesday.

**Comment by jharenza** [comment]
> @runjin326 , ok so Adam R. reached out to me, and he seems keen on seeing the project loaded. What will unblock this is that he has given input on preference to the issue of matched TARGET T/N RNA samples. He's like both loaded - so please add on more ability to differentiate between the 2 - a suggestion would be adding a `-T` and `-N`, but if there is already something easy that can help distinguish, feel free to use that. Feel free to reach out if you need clarification!

This should be able to be accomplished with standard TARGET nomenclature. @afarrel and/or @ewafula can you advise using the OCG codes? Thanks!

**Comment by runjin326** [comment]
@migbro - have not heard back from Alvin or Eric so I just went ahead and added `-N` and `-T` to the TARGET samples and they should be unique now. Attaching the file here: 
[histologies-formatted-id-added.txt](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/files/8400224/histologies-formatted-id-added.txt)

I haven't pushed the codes yet since I am not sure whether @jharenza would prefer using OCG codes. Let me know.

**Comment by ewafula** [comment]
@runjin326, @migbro, thus far, all the TARGET samples are unique among all experimental strategies expect for the upcoming v11 histologies file introduced by methylation array samples which we haven't resolved yet.

**Comment by ewafula** [comment]
> @runjin326, @migbro, thus far, all the TARGET samples are unique among all experimental strategies expect for the upcoming v11 histologies file introduced by methylation array samples which we haven't resolved yet.

Sorry, folks. This is a different issue that from the duplicate kids_First_Biospecimen_IDs in the histologies file that we discussed during the OT meeting.

**Comment by migbro** [comment]
> 



> @migbro - have not heard back from Alvin or Eric so I just went ahead and added `-N` and `-T` to the TARGET samples and they should be unique now. Attaching the file here: [histologies-formatted-id-added.txt](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/files/8400224/histologies-formatted-id-added.txt)
> 
> I haven't pushed the codes yet since I am not sure whether @jharenza would prefer using OCG codes. Let me know.

Thanks @runjin326  ! Just gave it a go. I don't think you're `-T` and `-N` additions made it into the sheet. However, I can probably toss it in there myself, just to see if it works

**Comment by migbro** [comment]
@runjin326 actually, let me clarify - it seems `TCGA` also has this matched T/N thing going on

**Comment by runjin326** [comment]
@migbro - oh ya, I only did it for TARGET not TCGA yet. Do you want me to do that? Or you would just give it a try? It might not work for TARGET though since some sample has multiple normal DNA samples and I had to use additional characters.

**Comment by migbro** [comment]
Also, in the off-chance that the samples set has like matched DNA T/N AND RNA T/N, I'd advocate for just adding `-N` to distinguish the normal, while also being able to match the DNA tumor to RNA tumor. Does that make sense? Like skip the `-T` part, unless you do it to the DNA tumor as well. Thanks!

**Comment by runjin326** [comment]
@migbro - I actually did not separate by DNA or RNA when adding -T and -N - if that answers your question.

**Comment by migbro** [comment]
Just saw your comment, I did give it a try, and it gave me all the warnings on TCGA, but none on TARGET. So, I think this solution worked for TARGET

**Comment by runjin326** [comment]
Sure - so question now is - do you want the same to be done for TCGA?

**Comment by runjin326** [comment]
@migbro - modified two things: 1) only added "-N" for samples; 2) did the same for TCGA as well.
[histologies-formatted-id-added.txt](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/files/8413151/histologies-formatted-id-added.txt)
Let me know whether it works now.

File: analyses/pedcbio-sample-name/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+## Add formatted sample id column for PedCBio upload
+
+**Module author:** Run Jin ([@runjin326](https://github.com/runjin326))
+
+Currently, for some of the samples, when multiple DNA or RNA specimens are associated with the same sample, there 
+is no column that would distinguish between different aliquots while still tieing DNA and RNA together.
```
Comment (migbro) [inline analyses/pedcbio-sample-name/README.md]:
```suggestion
is no column that would distinguish between different aliquots while still tying DNA and RNA together.
```

File: analyses/pedcbio-sample-name/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+or
+```
+bash run_add_name.sh
+```
+
+Input:
+- `input/cbtn_cbio_sample.csv`
+- `input/oligo_nation_cbio_sample.csv`
+- `input/dgd_cbio_sample.csv`
+- `input/x01_fy16_nbl_maris_cbio_sample.csvz`
```
Comment (migbro) [inline analyses/pedcbio-sample-name/README.md]: Extension looks weird - `csvz`

**Review by migbro — APPROVED** [review]
Just a couple minor comments, but given the code works "in the field," I think this is good to go. Thanks a lot for this!

File: analyses/pedcbio-sample-name/README.md (hunk 3 above)
Comment (runjin326) [reply analyses/pedcbio-sample-name/README.md]: thanks ! both comments modified now.

**Pull request merged**